lib/overcommit/hook_context/commit_msg.rb

```diff
@@ -31,6 +31,10 @@ def commit_message_file
       @args[0]
     end
 
+    def post_fail_message
+      "Failed commit message:\n" + commit_message_lines.join
+    end
```

Comment:
Let's define a default `post_fail_message` method in the base `HookContext` class so we don't need to do the `@context.respond_to?(:post_fail_message)` below. We can have a contract where returning `nil` results in nothing being printed.